### PR TITLE
fix row_del function in MutableDenseMatrix

### LINKED ISSUE #9876: A.inv() after A.row_del() and A.col_del() raises ValueError
```
>>> A = Matrix([[1, 2, 3], [5, 6, 7], [9, 10, 11]])
>>> A.col_del(-1)
>>> A.row_del(-1)
>>> A
Matrix([
[1, 2],
[5, 6]])
>>> A.inv()

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-61-e2932173dd42> in <module>()
----> 1 A.inv()

/home/gxyd/Public/sympy/sympy/matrices/matrices.py in inv(self, method, **kwargs)
    311         if method is not None:
    312             kwargs['method'] = method
--> 313         return self._eval_inverse(**kwargs)
    314 
    315     def inv_mod(self, m):

/home/gxyd/Public/sympy/sympy/matrices/dense.py in _eval_inverse(self, **kwargs)
    303             return diag(*r)
    304 
--> 305         M = self.as_mutable()
    306         if method == "GE":
    307             rv = M.inverse_GE(iszerofunc=iszerofunc)

/home/gxyd/Public/sympy/sympy/matrices/dense.py in as_mutable(self)
    597 
    598     def as_mutable(self):
--> 599         return self.copy()
    600 
    601     def __setitem__(self, key, value):

/home/gxyd/Public/sympy/sympy/matrices/matrices.py in copy(self)
    299 
    300     def copy(self):
--> 301         return self._new(self.rows, self.cols, self._mat)
    302 
    303     def trace(self):

/home/gxyd/Public/sympy/sympy/matrices/dense.py in _new(cls, *args, **kwargs)
    586     @classmethod
    587     def _new(cls, *args, **kwargs):
--> 588         rows, cols, flat_list = cls._handle_creation_inputs(*args, **kwargs)
    589         self = object.__new__(cls)
    590         self.rows = rows

/home/gxyd/Public/sympy/sympy/matrices/matrices.py in _handle_creation_inputs(cls, *args, **kwargs)
    213                 flat_list = args[2]
    214                 if len(flat_list) != rows*cols:
--> 215                     raise ValueError('List length should be equal to rows*columns')
    216                 flat_list = [cls._sympify(i) for i in flat_list]
    217 

ValueError: List length should be equal to rows*columns
````

While this works correctly
```
>>> B = Matrix([
[1, 2],
[5, 6]])
>>> B.inv()
Matrix([
[-3/2,  1/2],
[ 5/4, -1/4]])
````

### PR DESCRIPTION
**Pull request by leosartaj**
```
>>> A = Matrix([[1, 2, 3], [5, 6, 7], [9, 10, 11]])
>>> A.row_del(-1)
>>> A._mat
[1, 2, 3, 5, 6, 7, 9, 10, 11]
```

Last row was not deleted. This was due to how indexing is done in python. As pointed out by @jksuom [here](https://github.com/sympy/sympy/issues/9876#issuecomment-136021664).

eg.

```
>>> A._mat[-3:0]
[]
```

fixes #9876 

Ping @gxyd

### CODE REVIEW

File: sympy/matrices/dense.py [hunk 1]
```diff
@@ -881,7 +881,10 @@ def row_del(self, i):
         col_del
         """
         if i < -self.rows or i >= self.rows:
-            raise IndexError("Index out of range: 'i = %s', valid -%s <= i < %s" % (i, self.rows, self.rows))
+            raise IndexError("Index out of range: 'i = %s', valid -%s <= i"
+                             " < %s" % (i, self.rows, self.rows))
+        if i < 0:
+            i = self.rows - abs(i)
```
Comment (jksuom) [inline sympy/matrices/dense.py]: Or preferably ``i += self.rows``.

Comment (leosartaj) [inline sympy/matrices/dense.py]: That is cleaner. I will update shortly.

**Comment by leosartaj** [comment]
I have addressed your comment @jksuom.

**Comment by gxyd** [comment]
looks good +1.

**Comment by jksuom** [comment]
Thanks, merging.

**Comment by leosartaj** [comment]
Thanks @jksuom @gxyd.

**Pull request merged**